lib/monkey_patch_activerecord.rb

```diff
@@ -178,7 +177,11 @@ def _update_record(values, id, id_was) # :nodoc:
       # ****** BEGIN PARTITIONED PATCH ******
       if @klass.respond_to?(:dynamic_arel_table)
         using_arel_table = @klass.dynamic_arel_table(Hash[*values.inject([]) { |result, (k, v)| result += [k.name, v] }])
```

Comment:
could we please use:
`*values.flat_map { |k,v| [k.name, v] }`
instead of `inject([])` ?